src/main/java/com/hubspot/jinjava/lib/filter/RegexReplaceFilter.java

```diff
@@ -0,0 +1,50 @@
+package com.hubspot.jinjava.lib.filter;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+
+import com.hubspot.jinjava.doc.annotations.JinjavaDoc;
+import com.hubspot.jinjava.doc.annotations.JinjavaParam;
+import com.hubspot.jinjava.doc.annotations.JinjavaSnippet;
+import com.hubspot.jinjava.interpret.InterpretException;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+
+@JinjavaDoc(
+        value = "Return a copy of the value with all occurrences of a matched regular expression replaced with a new one. " +
+                "The first argument is the regular expression to be matched, the second is the replacement string",
+        params = {
+                @JinjavaParam(value = "s", desc = "Base string to find and replace within"),
+                @JinjavaParam(value = "regex", desc = "The regular expression that you want to match and replace"),
+                @JinjavaParam(value = "new", desc = "The new string that you replace the matched substring")
+        },
+        snippets = {
+                @JinjavaSnippet(
+                        code = "{{ \"It costs $300\"|regex_replace(\"[^a-zA-Z]\", \"\") }}",
+                        output = "Itcosts")
+        })
+public class RegexReplaceFilter implements Filter {
+
+    @Override
+    public String getName() {
+        return "regex_replace";
+    }
+
+    @Override
+    public Object filter(Object var, JinjavaInterpreter interpreter,
+                         String... args) {
+
+        if (var == null) {
+            return null;
+        }
+        if (args.length < 2) {
+            throw new InterpretException("filter " + getName() + " requires two string args");
+        }
+
+        String s = (String) var;
+        String toReplace = args[0];
+        String replaceWith = args[1];
+
+        return s.replaceAll(toReplace, replaceWith);
```

Comment:
This may throw a `PatternSyntaxException` we should catch and add to the template errors.